packages/core/src/amazonqFeatureDev/storages/chatSession.ts

```diff
@@ -20,12 +23,14 @@ export class ChatSessionStorage {
     }
 
     public async getSession(tabID: string): Promise<Session> {
-        const sessionFromStorage = this.sessions.get(tabID)
-        if (sessionFromStorage === undefined) {
-            // If a session doesn't already exist just create it
-            return this.createSession(tabID)
-        }
-        return sessionFromStorage
+        return this.lock.acquire(tabID, async () => {
```

Comment:
I'm curious if we can have `this.lock.acquire` as decorator?

Reply:
That is still a typescript's experimental feature, correct?

Reply:
Would that make the code clearer, or less obvious?